packages/@okta/vuepress-site/docs/guides/validate-access-tokens/overview/java/accesstoken.md

```diff
@@ -1,17 +1,59 @@
 See the instructions for the [Okta JWT Verifier for Java](https://github.com/okta/okta-jwt-verifier-java).
 
-## Verify the Token Signature
-
-You verify the Access or ID token's signature by matching the key that was used to sign in with one of the keys that you retrieved from your Okta Authorization Server's JWK endpoint. Specifically, each public key is identified by a `kid` attribute, which corresponds with the `kid` claim in the Access or ID token header.
-
-If the `kid` claim doesn't match, it's possible that the signing keys have changed. Check the `jwks_uri` value in the Authorization Server metadata and try retrieving the keys again from Okta.
-
-### Verify the Claims
-
-You should verify the following:
-
-- The `iss` (issuer) claim matches the identifier of your Okta Authorization Server.
-- Verify that the `aud` (audience) claim is the value configured in the Authorization Server.
-- The `exp` (expiry time) claim is the time at which this token will expire, expressed in Unix time. You should make sure that this has not already passed.
-
+### Required Libraries
+
+For validating a JWT, you will need the Okta JWT Verifier for Java library.
+
+Include the below lines in your Maven pom.xml:
+
+```xml
+<dependency>
+   <groupId>com.okta.jwt</groupId>
+   <artifactId>okta-jwt-verifier</artifactId>
+   <version>${okta-jwt.version}</version>
+</dependency>
+  
+<dependency>
+  <groupId>com.okta.jwt</groupId>
+  <artifactId>okta-jwt-verifier-impl</artifactId>
+  <version>${okta-jwt.version}</version>
+  <scope>runtime</scope>
+</dependency>
+```
+
+### Validate Access Token
+
+Create the Okta JWT Verifier using `JwtVerifiers` class:
+
+```java
+AccessTokenVerifier jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+    .setIssuer("https://{yourOktaDomain}/oauth2/default")
+    .setAudience("api://default")                // defaults to 'api://default'
+    .setConnectionTimeout(Duration.ofSeconds(1)) // defaults to 1s
+    .setReadTimeout(Duration.ofSeconds(1))       // defaults to 1s
+    .build();
+```
+
+This helper class configures a JWT parser with the details found via the [OpenID Connect discovery endpoint](https://openid.net/specs/openid-connect-discovery-1_0.html).  The public keys used to validate the JWTs will also be retrieved and cached automatically.
```

Comment:
```suggestion
This helper class configures a JWT parser with the details found through the [OpenID Connect discovery endpoint](https://openid.net/specs/openid-connect-discovery-1_0.html).  The public keys used to validate the JWTs will also be retrieved and cached automatically.
```

Reply:
Would it be more appropriate to link to the docs for our OpenID discovery endpoint? Or is this not using our endpoint?

Reply:
Not sure on this one. @bdemers your thoughts?

Reply:
+1, this content is a little dated, so updating to point to Okta links would be better (and adds more cross-linking)

Reply:
@jakubvul-okta just to double check - https://developer.okta.com/docs/reference/api/oidc/#well-knownoauth-authorization-server is the right link to reference from here?